test/batched.js

```diff
@@ -346,3 +347,44 @@ test('POST batched query with a resolver which succeeds and a resolver which thr
 
   t.same(JSON.parse(res.body), [{ data: { add: 3 } }, { data: null, errors: [{ message: 'Internal Server Error' }] }])
 })
+
+test('POST batched query has an individual context for each operation', async (t) => {
+  const app = Fastify()
+
+  const contextSpy = sinon.spy()
+
+  const schema = `
+      type Query {
+        test: String
+      }
+    `
+
+  const resolvers = {
+    test: (_, ctx) => contextSpy(ctx.operationId, ctx.operationsCount, ctx.__currentQuery)
+  }
+
+  app.register(GQL, {
+    schema,
+    resolvers,
+    allowBatchedQueries: true
+  })
+
+  await app.inject({
+    method: 'POST',
+    url: '/graphql',
+    body: [
+      {
+        operationName: 'TestQuery',
+        query: 'query TestQuery { test }'
+      },
+      {
+        operationName: 'DoubleQuery',
+        query: 'query DoubleQuery { test }'
+      }
+    ]
+  })
+
+  sinon.assert.calledTwice(contextSpy)
+  sinon.assert.calledWith(contextSpy, 0, 2, sinon.match(/TestQuery/))
+  sinon.assert.calledWith(contextSpy, 1, 2, sinon.match(/DoubleQuery/))
```

Comment:
why are we not testing that operationId and operationCount are also passed around properly?

Reply:
I may not be understanding what you're saying, because I believe that's what we're doing here: when calling the resolver we're reading the `operationId` and `operationsCount` attributes from the context, and checking that they've been passed as expected: a call with `operationId === 0`, another call with `operationId === 1`, both of them with `operationsCount === 2`, and each one of them with the correct query contents.

Reply:
sorry I read that wrong, I thought those were the number of the call and the position of the argument for some reason!

Reply:
Haha, I feel you: I even had to double check the spy parameters when writing that assertion.